### PR TITLE
test(core): add ut for app

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add supertest to test the Koa server.

## Linear Issue Reference

LOG-1540

## Testing

UT passed

@logto-io/eng

### CODE REVIEW

File: packages/core/src/app/init.test.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import Koa from 'koa';
+import request from 'supertest';
+
+import { envVariablesSetUp } from '@/utils/test-utils';
+
+/**
+ * Need to mock env variables ahead
+ */
+
+envVariablesSetUp();
```
Comment (wangsijie) [inline packages/core/src/app/init.test.ts:10]: can we move it to `jest.setup`?

File: packages/core/src/app/init.ts [hunk 2]
```diff
@@ -7,18 +7,13 @@ import koaLogger from 'koa-logger';
...
 import koaI18next from '@/middleware/koa-i18next';
-import koaOIDCErrorHandler from '@/middleware/koa-oidc-error-handler';
-import koaSlonikErrorHandler from '@/middleware/koa-slonik-error-handler';
 import koaUIProxy from '@/middleware/koa-ui-proxy';
 import koaUserLog from '@/middleware/koa-user-log';
 import initOidc from '@/oidc/init';
 import initRouter from '@/routes/init';
 
-export default async function initApp(app: Koa): Promise<void> {
+export default async function initApp(app: Koa) {
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/core/src/app/init.ts:15]: I'm OK with that.

**Review by wangsijie — APPROVED** [review]
overall LGTM

File: packages/core/src/app/init.test.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/app/init.test.ts:10]: Use to add it as a `jest.setup`. But not confident whether we need to set the env variables for all the ut cases.  So decide to keep it in file for now. Move to jest.setup till the next requirement.

File: packages/core/src/app/init.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gao-sun) [reply packages/core/src/app/init.ts:15]: it'll be better to explicitly write return type for `export` functions

File: packages/core/src/app/init.test.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/app/init.test.ts:10]: ok

File: packages/core/src/utils/test-utils.ts [hunk 3]
```diff
@@ -21,3 +21,15 @@ export const createTestPool = <T extends QueryResultRowType>(
...
+    'LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlDV2dJQkFBS0JnR3pLendQcVp6Q3dncjR5a0U1NTN2aWw3QTZYM2l1VnJ3TVJtbVJDTVNBL3lkUm04bXA1CjlHZUYyMlRCSVBtUEVNM29Lbnk4KytFL2FDRnByWXVDa0loREhodVR5N1diT25nd3kyb3JpYnNEQm1OS3FybTkKM0xkYWYrZm1aU2tsL0FMUjZNeUhNV2dTUkQrbFhxVnplNFdSRGIzVTlrTyt3RmVXUlNZNmlRL2pBZ01CQUFFQwpnWUJOZkczUjVpUTFJNk1iZ0x3VGlPM3N2NURRSEE3YmtETWt4bWJtdmRacmw4TlRDemZoNnBiUEhTSFVNMUlmCkxXelVtMldYanFzQUZiOCsvUnZrWDh3OHI3SENNUUdLVGs0ay9adkZ5YUhkM2tIUXhjSkJPakNOUUtjS2NZalUKRGdnTUVJeW5PblNZNjJpWEV6RExKVTJEMVUrY3JEbTZXUTVHaG1NS1p2Vnl3UUpCQU1lcFBFV2gwakNDOEdmQwpQQU1yT1JvOHJYeHYwVEdXNlJWYmxad0ppdjhNeGZacnpZT1cwZUFPek9IK0ZRWE90SjNTdUZONzdEcVQ5TDI3CmN2M3QySkVDUVFDTGZZeVl2ZUg0UnY2bnVET0RnckkzRUJHMFNJbURHcC94UUV2NEk5Z0hrRFF0aFF4bW5xNTEKZ1QxajhFN1lmRHEwMTkvN2htL3dmMXNzMERQNkpic3pBa0JqOEUzKy9MVGRHMjJDUWpNUDB2N09KemtmWkVqdAo3WC9WOVBXNkdQeStGWUt4aWR4ZzFZbFFBWmlFTms0SGppUFNLN3VmN2hPY2JwcStyYWt0ZVhSQkFrQmhaaFFECkh5c20wbVBFTnNGNWhZdnRHTUpUOFFaYnpmNTZWUnYyc3dpSUYyL25qT3hneDFJbjZFczJlamlEdnhLNjdiV1AKQ29zbEViaFhMVFh0NStTekFrQjJQOUYzNExubE9tVjh4Zjk1VmVlcXNPbDFmWWx2Uy9vUUx1a2ZxVkJsTmtzNgpzdmNLVDJOQjlzSHlCeE8vY3Zqa0ZpWXdHR2MzNjlmQklkcDU1S2IwCi0tLS0tRU5EIFJTQSBQUklWQVRFIEtFWS0tLS0t';
```
Comment (gao-sun) [inline packages/core/src/utils/test-utils.ts:27]: should we just random a key here?

Comment (simeng-li) [reply packages/core/src/utils/test-utils.ts:27]: tested, won't work, the input value must be a strict base64 private key

Comment (gao-sun) [reply packages/core/src/utils/test-utils.ts:27]: yes i mean the value should be generated per test - but not a blocker

File: packages/core/src/app/init.test.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    koaErrorHandler,
+    koaI18next,
+    koaOIDCErrorHandler,
+    koaSlonikErrorHandler,
+    koaUIProxy,
+    koaUserLog,
+  ];
+  const initMethods = [initRouter, initOidc];
+
+  const middleWareSpys = middleWares.map((module) => jest.spyOn(module, 'default'));
```
Comment (gao-sun) [inline packages/core/src/app/init.test.ts:39]: middleware is an uncountable noun

**Review by gao-sun — APPROVED** [review]

File: packages/core/src/app/init.test.ts (hunk 4 above)
Comment (simeng-li) [reply packages/core/src/app/init.test.ts]: updated

File: packages/core/src/utils/test-utils.ts (hunk 3 above)
Comment (simeng-li) [reply packages/core/src/utils/test-utils.ts:27]: got you. will update later

**Pull request merged**